ts_src/psbt.ts

```diff
@@ -1743,8 +1754,8 @@ function getTaprootHashesForSig(
 
   const hashes = [];
   if (input.tapInternalKey && !tapLeafHashToSign) {
-    const outputKey = tweakInternalPubKey(inputIndex, input);
-    if (toXOnly(pubkey).equals(outputKey)) {
+    const outputKey = getPrevoutTaprootKey(inputIndex, input, cache);
```

Comment:
This part is breaking one of the test vectors in the unit test for signing.

The test vector says `Sign PSBT with 3 inputs [P2PKH, P2TR (key-path), P2WPKH]` and at this point.

This is confusing...

```
ECPair keypair (pubkey)   4fef5c5163bea69a93e74a59672bbeb081837077cb94cfa6e481a5cf00d8ab18
tweak(tapInternalKey)     4fef5c5163bea69a93e74a59672bbeb081837077cb94cfa6e481a5cf00d8ab18

taproot prevOutput key    9421e734b0f9d2c467ea7dd197c61acb4467cdcbc9f4cb0c571f8b63a5c40cae
input.tapInternalKey      9421e734b0f9d2c467ea7dd197c61acb4467cdcbc9f4cb0c571f8b63a5c40cae
```

The tapInternalKey and the prevout script key are the same, and the WIF for the keypair is the self-tweaked version of tapInternalKey/prevOutputKey.

I think this test vector is incorrect...

Reply:
```
> p.data.inputs[1].witnessUtxo
{
  script: <Buffer 51 20 94 21 e7 34 b0 f9 d2 c4 67 ea 7d d1 97 c6 1a cb 44 67 cd cb c9 f4 cb 0c 57 1f 8b 63 a5 c4 0c ae>,
  value: 67000
}
> p.data.inputs[1].tapInternalKey
<Buffer 94 21 e7 34 b0 f9 d2 c4 67 ea 7d d1 97 c6 1a cb 44 67 cd cb c9 f4 cb 0c 57 1f 8b 63 a5 c4 0c ae>
```